explorer/data/prelude.carbon

```diff
@@ -86,6 +90,184 @@ impl String as EqWith(Self) {
   }
 }
 
+// ----------------------
+// COMPARE
+// ----------------------
+
+choice Ordering {
+  Less,
+  Equivalent,
+  Greater,
+  Incomparable
+}
+
+interface CompareWith(U:! Type) {
+  fn Compare[me: Self](u: U) -> Ordering;
+  // TODO: Add `default fn` for Less, LessOrEquivalent, Greater, and GreaterOrEquivalent once it's available.
+}
+// TODO: constraint Ordered { ... }
+
+impl i32 as CompareWith(Self) {
+  fn Compare[me: Self](other: Self) -> Ordering {
+    var comp: i32 = __intrinsic_int_compare(me, other);
+    if (comp == -1) {
+      return Ordering.Less();
+    }
+    if (comp == 0) {
+      return Ordering.Equivalent();
+    }
+    if (comp == 1) {
+      return Ordering.Greater();
+    }
+    return Ordering.Incomparable();
+
+  }
+}
+
+impl String as CompareWith(Self) {
+  fn Compare[me: Self](other: Self) -> Ordering {
+    var comp: i32 = __intrinsic_str_compare(me, other);
+    if (comp == -1) {
+      return Ordering.Less();
+    }
+    if (comp == 0) {
+      return Ordering.Equivalent();
+    }
+    if (comp == 1) {
+      return Ordering.Greater();
+    }
+    return Ordering.Incomparable();
+  }
+}
+
+
+interface LessWith(U:! Type) {
+  fn Less[me: Self](other: U) -> Bool;
+}
+
+interface LessEqWith(U:! Type) {
+  fn LessEq[me: Self](other: U) -> Bool;
+}
+
+interface GreaterWith(U:! Type) {
+  fn Greater[me: Self](other: U) -> Bool;
+}
+
+interface GreaterEqWith(U:! Type) {
+  fn GreaterEq[me: Self](other: U) -> Bool;
+}
+
+
+
+
+impl i32 as LessWith(Self) {
+  fn Less[me: Self](other: Self) -> Bool {
+    var comp: Ordering =  me.(CompareWith(i32).Compare)(other);
+    match(comp){
+         case Ordering.Less() =>{
+          return true;
+         }
+     }
+     return false;
```

Comment:
Style-wise, I think there are some slight differences between what's written and what I'd expect. To try to illustrate:

```suggestion
    var comp: Ordering = me.(CompareWith(i32).Compare)(other);
    match (comp) {
      case Ordering.Less() => {
        return true;
      }
    }
    return false;
```

Would you prefer to merge as-is and I can just plan a formatting pass separately, or is this something you'd like to clean up? (I wanted to note highlight it though if you're planning on regular contributions, but really this has me back to wondering if we can figure out something fast that could do formatting for us)